app/controllers/vendor_api/applications_controller.rb

```diff
@@ -13,19 +13,28 @@ def show
   private
 
     def serialized_application_choices_data
-      json_data = get_application_choices_for_provider_since(since: since_param).map do |application_choice|
-        ApplicationPresenter.new(version_number, application_choice).serialized_json
-      end
-
-      %({"data":[#{json_data.join(',')}]})
+      MultipleApplicationsPresenter.new(version_number, get_application_choices_for_provider_since(since: since_param), pagination_params).serialized_applications_data
     end
 
     def get_application_choices_for_provider_since(since:)
       application_choices_visible_to_provider
         .where('application_choices.updated_at > ?', since)
-        .find_each(batch_size: 500)
```

Comment:
why did we remove finding in batches?

Reply:
We run into issues with the pagy implementation:
```
NoMethodError:
       undefined method `order' for #<Enumerator:0x00007ff3a9ada918>
```

Reply:
Just thought – this will impact v1.0 as well. Will work on a change now